src/argilla/server/models/models.py

```diff
@@ -170,13 +173,22 @@ def __repr__(self):
 DatasetStatusEnum = SAEnum(DatasetStatus, name="dataset_status_enum")
 
 
+def _updated_at_current_value(context: DefaultExecutionContext) -> datetime:
+    return context.get_current_parameters(isolate_multiinsert_groups=False)["updated_at"]
+
+
 class Dataset(DatabaseModel):
     __tablename__ = "datasets"
 
     name: Mapped[str] = mapped_column(index=True)
     guidelines: Mapped[Optional[str]] = mapped_column(Text)
     status: Mapped[DatasetStatus] = mapped_column(DatasetStatusEnum, default=DatasetStatus.draft, index=True)
     workspace_id: Mapped[UUID] = mapped_column(ForeignKey("workspaces.id", ondelete="CASCADE"), index=True)
+    inserted_at: Mapped[datetime] = mapped_column(default=datetime.utcnow)
+    updated_at: Mapped[datetime] = mapped_column(default=inserted_at_current_value, onupdate=datetime.utcnow)
+    last_activity_at: Mapped[datetime] = mapped_column(
```

Comment:
What would be the value for existing datasets? Do we need to do an extra migration step?

Reply:
Yes, I totally miss that. We can copy the value from `updated_at` or set it as optional.

Reply:
Done. I updated the migration to use `updated_at` value to populate the new column.